main.go

```diff
@@ -266,68 +266,123 @@ func findPointParallel(ctx context.Context, workers int, p0 *edwards25519.Point,
 }
 
 func findBatchPoint(ctx context.Context, p0 *edwards25519.Point, skip uint64, batchSize int, test func([]byte) bool, limit uint64) (uint64, bool) {
+	if batchSize < 0 || batchSize%2 != 0 {
+		panic("batchSize must be non-negative and even")
+	}
+
 	skipOffset := new(edwards25519.Point).ScalarMult(scalarFromUint64(skip), pointOffset)
 	p := new(edwards25519.Point).Add(p0, skipOffset)
 
 	n := skip
 
-	ua := make([]field.Element, batchSize+1)
-	ub := make([]field.Element, batchSize+1)
-	u := make([]field.Element, batchSize+1)
+	ua := make([]field.Element, batchSize+2)
+	ub := make([]field.Element, batchSize+2)
+	u := make([]field.Element, batchSize+2)
 	scratch := make([][]field.Element, 2)
 
 	for i := range scratch {
-		scratch[i] = make([]field.Element, batchSize+1)
+		scratch[i] = make([]field.Element, batchSize+2)
 	}
 
-	// offsets[i] = i * pointOffset
-	offsets := make([]affine, batchSize)
-	offsets[0].zero()
+	// offsets[i] = (i+1) * pointOffset
+	offsets := make([]affine, batchSize/2)
 	poi := new(edwards25519.Point).Set(pointOffset)
-	for i := 1; i < len(offsets); i++ {
+	for i := range batchSize/2 - 1 {
 		offsets[i].fromP3(poi)
 		poi.Add(poi, pointOffset)
 	}
-	batchOffset := poi // batchSize * pointOffset
+	// batchOffset = (batchSize+1) * pointOffset
+	batchOffset := new(edwards25519.Point).Set(pointOffset)
+	if batchSize > 0 {
+		offsets[batchSize/2-1].fromP3(poi)
+		batchOffset.Add(batchOffset, poi)
+		batchOffset.Add(batchOffset, poi)
+		p.Add(p, poi)
+	}
 
-	// Starting point for the current batch
+	// Center point for the current batch
 	pa := new(affine).fromP3(p)
 
 	var bm [32]byte
-	// Loop body complexity: (6M + 4A)*batchSize + 278M + 9A
+	// One iteration tests batchSize+1 points of the
+	// batch = {p − batchSize/2*pointOffset, ... , p − pointOffset, p, p + pointOffset, ... , p + batchSize/2*pointOffset}
+	//
+	// Complexity: (5M + 4A)*batchSize + 282M + 11A
```

Comment:
This is amortized 5 multiplications per point: 1M for addition and 4M for inversion.

Reducing this further requires a breakthrough like discovery of a better addition or inversion algorithms or a completely different approach.

Reply:
Well, it turned out that using Montgomery coordinates requires only 3.5 multiplications, see https://github.com/AlexanderYastrebov/vanity25519/